src/test/java/com/training/assignment/tests/AssignmentTest.java

```diff
@@ -139,4 +142,32 @@ public void testOopsConcept() {
         softAssert.assertThat(director.getWorkDetail())
                 .as("Work of actor and director differs").isEqualTo("Acting Directing");
     }
+
+    @Test
+    public void testAbstractInterfaceConcept() {
+        Employee employee = new Employee();
+
+        Yes1 yes1 = new Yes1();
+
+        Yes2 yes2 = new Yes2();
+
+        String firstName="Shrikant";
+        String lastName="Sajjanshetti";
+        String middleName="Umakant";
+
+        softAssert.assertThat(employee.getName(firstName))
+                .as("Employee name doesn't match,try something else").isEqualTo("Shrikant");
+
+        softAssert.assertThat(employee.getName(firstName, lastName))
+                .as("Employee name doesnt match,try something else").isEqualTo("Shrikant Sajjanshetti");
+
+        softAssert.assertThat(employee.getName(firstName, lastName, middleName))
+                .as("Employee name doesnt match,try something else").isEqualTo("Shrikant Sajjanshetti Umakant");
+
+        softAssert.assertThat(yes1.getSum(3))
+                .as("Sum of three number is incorrect").isEqualTo(33);
+
+        softAssert.assertThat(yes2.getSum(5))
```

Comment:
Direclty use, softAssert.assertThat(new Yes1().getSum(3))
                .as("Sum of three number is incorrect").isEqualTo(33);

And of below code too